releases/release-5.1.1.md

```diff
@@ -0,0 +1,113 @@
+---
+title: TiDB 5.1.1 Release Notes
+---
+
+# TiDB 5.1.1 Release Notes
+
+Release Date: July 30, 2021
+
+TiDB version: 5.1.1
+
+## Compatibility change
+
++ TiDB
+
+    - For users upgrading from TiDB 4.0, the value of tidb_multi_statement_mode is now OFF. It is recommended to use the multi-statement feature of your client library instead, see the documentation on tidb_multi_statement_mode for additional details. [#25751](https://github.com/pingcap/tidb/pull/25751)
+
+## Feature enhancements
+
+## Improvements
+
++ TiDB
+
+    - avoid alloc for paramMarker in buildValuesListOfInsert [#26076](https://github.com/pingcap/tidb/pull/26076)
+    - planner: support stable result mode [#25995](https://github.com/pingcap/tidb/pull/25995)
+
++ TiKV
+
+    - separate read write ready to reduce read latency [#10592](https://github.com/tikv/tikv/pull/10592)
+
+## Bug fixes
+
++ TiDB
+
+    - amend transactions correctly when "modify column" needs reorg data with tidb_enable_amend_pessimistic_txn=on. [#26273](https://github.com/pingcap/tidb/pull/26273)
+    - fix incompatible last_day func behavior in sql mode [#26001](https://github.com/pingcap/tidb/pull/26001)
+    - Make sure limit outputs no more columns than its child [#25980](https://github.com/pingcap/tidb/pull/25980)
+    - Fix the issue that committing pessimistic transactions may report write-conflict errors. [#25973](https://github.com/pingcap/tidb/pull/25973)
+    - planner: handle other-conditions from subqueries correctly when constructing IndexJoin [#25819](https://github.com/pingcap/tidb/pull/25819)
+    - Fix the bug that successful optimistic transactions may report commit errors. [#25803](https://github.com/pingcap/tidb/pull/25803)
+    - Fix incorrect result of set type for merge join [#25695](https://github.com/pingcap/tidb/pull/25695)
+
```

Comment:
- Fix the bug that index keys in a pessimistic transaction may be repeatedly committed. [#26482](https://github.com/pingcap/tidb/pull/26482)
    - planner: fix the risk of integer overflow when locating partitions [#26471](https://github.com/pingcap/tidb/pull/26471)
    - ddl: fix cast date as timestamp will write invalid value [#26395](https://github.com/pingcap/tidb/pull/26395)
    - Fix copt-cache metrics, it will display the number of hits/miss/evict on Grafana. [#26344](https://github.com/pingcap/tidb/pull/26344)
    - fix the bug of annoying logs caused by telemetry [#26284](https://github.com/pingcap/tidb/pull/26284)
    - fix a bug on the query range of prefix index [#26262](https://github.com/pingcap/tidb/pull/26262)
    - Fix the issue that concurrently truncating the same partition hangs DDL. [#26239](https://github.com/pingcap/tidb/pull/26239)
    - planner/core: fix duplicate enum items [#26202](https://github.com/pingcap/tidb/pull/26202)
    - executor: fix a bug that cte.iterOutTbl did not close correctly [#26148](https://github.com/pingcap/tidb/pull/26148)
    - load: fix load data with non-utf8 can succeed [#26144](https://github.com/pingcap/tidb/pull/26144)
    - fix unsigned int window function error [#26027](https://github.com/pingcap/tidb/pull/26027)
    - Fix the issue that TiDB may panic when resolving async-commit locks. [#25862](https://github.com/pingcap/tidb/pull/25862)
    - Make Stale Read fully support prepare statement [#25800](https://github.com/pingcap/tidb/pull/25800)
    - Fix the issue that ODBC-styled literal(like {d '2020-01-01'}...) cannot be used as the expression. [#25578](https://github.com/pingcap/tidb/pull/25578)    
    - fix the bug about unnecessary error when run tidb only [#25555](https://github.com/pingcap/tidb/pull/25555)